### PR TITLE
Update the SDK.

### PR DESCRIPTION
**Pull request by pixlwave**
Nothing exciting here, just fixing a couple of API breaks.

### CODE REVIEW

**Review by stefanceriu — APPROVED** [review]

**Comment by pixlwave** [comment]
I've fixed all the crashes and refactored the Voice Message player code somewhat so all the tests should be passing now.

File: ElementX/Sources/Services/Audio/Player/AudioPlayer.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ private enum InternalAudioPlayerState {
 }
 
 class AudioPlayer: NSObject, AudioPlayerProtocol {
-    var mediaSource: MediaSourceProxy?
+    var sourceURL: URL?
```
Comment (pixlwave) [inline ElementX/Sources/Services/Audio/Player/AudioPlayer.swift:25]: Turns out this is needed, but externally via the protocol so I reduced it to just the URL which is enough to identify the loaded file by.

File: ElementX/Sources/Mocks/MediaProviderMock.swift [hunk 2]
```diff
@@ -18,7 +18,8 @@ extension MediaProviderMock {
                 return nil
             }
             
-            if mediaSource?.url == .picturesDirectory {
+            // At some stage it would be nice to return different images, but for now they can be the same.
```
Comment (stefanceriu) [inline ElementX/Sources/Mocks/MediaProviderMock.swift:21]: 👍

File: ElementX/Sources/Other/Extensions/URL.swift [hunk 3]
```diff
@@ -100,4 +100,12 @@ extension URL: @retroactive ExpressibleByStringLiteral {
         }
         return nil
     }
+    
+    // MARK: Mocks
```
Comment (stefanceriu) [inline ElementX/Sources/Other/Extensions/URL.swift:104]: Nice!

File: ElementX/Sources/Services/Audio/Player/AudioPlayerProtocol.swift [hunk 4]
```diff
@@ -12,6 +12,17 @@ enum AudioPlayerError: Error {
     case genericError
 }
 
+// There used to be a MediaPlayerProtocol that AudioPlayerProtocol inherited from.
+// This should be called something else but we already have an AudioPlayerState,
+// AudioPlayerPlaybackState and InternalAudioPlayerState so who knows what to call this.
```
Comment (stefanceriu) [inline ElementX/Sources/Services/Audio/Player/AudioPlayerProtocol.swift:17]: 🤦‍♂️

File: ElementX/Sources/Services/Audio/Player/AudioPlayerProtocol.swift [hunk 5]
```diff
@@ -22,8 +33,21 @@ enum AudioPlayerAction {
     case didFailWithError(error: Error)
 }
 
-protocol AudioPlayerProtocol: MediaPlayerProtocol {
+protocol AudioPlayerProtocol: AnyObject {
```
Comment (stefanceriu) [inline ElementX/Sources/Services/Audio/Player/AudioPlayerProtocol.swift:36]: 👌

**Review by stefanceriu — APPROVED** [review]
Sorry you had to go through this but super nice cleanup and just look at those snapshots! 🤩 😁

**Pull request merged**